### ISSUE TITLE
Slow/jerky animation when resizing due to cells reloading.

### ISSUE THREAD

**Issue by swaterfall** [jwilling/JNWCollectionView#117]
Hey,
If you resize the window frame all the cells get reloaded (cellForItemAtIndexPath) even those that are visible before and after the resize, so if you have a lot of cells visible the animation can be slow and judder. 

Is it possible to call cellForItemAtIndexPath only on any new cells that are introduced with the resizing? Or is there another solution?

**Comment by rharter** [comment]
I've tried creating a `JNWCollectionViewGridLayout` subclass that simply overrides `shouldInvalidateLayoutForBoundsChange` with this:

```objective-c
- (BOOL)shouldInvalidateLayoutForBoundsChange:(CGRect)newBounds; {
    NSArray *oldItems = _items;
    _items = [self indexPathsForItemsInRect:newBounds];
    return ![_items isEqualToArray:oldItems];
}
```

That certainly seems to make it not reload each cell until a column boundary is crossed, but it also doesn't do any adjustment while dragging, so columns don't change until you actually cut on off entirely.  It also doesn't work when dragging the window larger, horizontally.

I was hoping that the recycling of views would be decoupled from the laying out of views, i.e. you could have one pass that determines which views to get rid of and which can stay, instantiating new ones as needed, and a second pass that actually adjusts the layout (size/positioning) of the views.  Is that not the intended architecture?

**Comment by jwilling** [comment]
@rharter I'll be frank with you: it has been several years since I wrote that code, and I sadly do not remember my reasoning for always discarding visible views when the bounds of the collection view has changed. I would suggest forking this repository and implementing the change I outlined above to see if there are any adverse effects when resizing. I do remember it was implemented to solve some edge case, but sadly I no longer remember what it was. I suppose I should have documented it better, but hindsight is 20/20.
